homeassistant/components/switch/config_flow.py

```diff
@@ -9,20 +9,24 @@
 
 from .const import DOMAIN
 
-DATA_SCHEMA = vol.Schema(
-    {
-        vol.Required("entity_id"): selector.selector({"entity": {"domain": "switch"}}),
-        vol.Required("name"): selector.selector({"text": {}}),
-    }
-)
+STEPS = {
+    "init": vol.Schema(
+        {
+            vol.Required("entity_id"): selector.selector(
+                {"entity": {"domain": "switch"}}
+            ),
+            vol.Required("name"): selector.selector({"text": {}}),
```

Comment:
Let's not ask name. We can add a unique ID based on the config entry and let the entity registry handle rename.

Reply:
I had the same discussion with @frenck, he had the same opinion.
Updated to remove the name and also drop the options flow since it doesn't make sense to reconfigure the source switch.

If we don't allow the user to input an initial name of the config entry when it's created, how can they be distinguished after creation? Should we name them the config entry + its light entity the same as the switch entity, with fallback to the switch's `entity_id` if it's not in the state machine?

Reply:
Updated to name the config entry + its light entity the same as the switch entity, with fallback to the switch's `object_id`.